testcase/unittest/blockchain/transactions/test_transaction_verifier_cache.py

```diff
@@ -16,29 +15,16 @@
 tx_versioner = TransactionVersioner()
 
 
+@pytest.mark.parametrize("tx_version", [
+    v2.version, v3.version
+])
 class TestTxCache:
     target_attrs = ["hash", "signature"]
 
-    @pytest.fixture
-    def tx(self):
-        """Transaction fixture"""
-        tb = TransactionBuilder.new(version="0x3", type_=None, versioner=tx_versioner)
-        tb.step_limit = 1000000
-        tb.value = 100000
-        tb.signer = Signer.from_prikey(os.urandom(32))
-        tb.to_address = ExternalAddress(os.urandom(20))
-        tb.nid = 3
-        tb.nonce = random.randint(0, 100000)
-        tb.data = "test"
-        tb.data_type = "message"
-
-        transaction = tb.build()
-
-        return transaction
-
     @pytest.mark.parametrize("target_attr", target_attrs)
-    def test_no_cache_attr_in_tx_if_not_verified(self, tx, target_attr):
+    def test_no_cache_attr_in_tx_if_not_verified(self, tx_version, tx_factory, target_attr):
         """Check that Transaction has no attribute until the verification func has been invoked"""
+        tx = tx_factory(tx_version=tx_version)
         if "cache" in dir(tx):
```

Comment:
```python
if any("cache" in attr for attr in dir(tx):
```

Reply:
I was definitely wrong. This test catches only `Transaction.cache` attribute.